Data Governance KIT Documentation

dg-tsy opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description
<!-- 
Please describe your PR: 
- What does this PR introduce? 
- Does it fix a bug? 
- Does it add a new feature?
- Is it enhancing documentation?
-->

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [ ] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [ ] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

dg-tsy commented:
Hello Max,

I have created the GitHub issue for release checks - Data Governance Kit - Release Checks #542<https://github.com/eclipse-tractusx/sig-release/issues/542>

Kind regards / Viele Grüße
Deepak Gupta
Product Owner
Data Governance KIT
Catena-X

***@***.***

From: Maximilian Ong ***@***.***>
Sent: Wednesday, February 21, 2024 2:17 PM
To: eclipse-tractusx/eclipse-tractusx.github.io ***@***.***>
Cc: Gupta, Deepak ***@***.***>; Author ***@***.***>
Subject: Re: [eclipse-tractusx/eclipse-tractusx.github.io] Data Governance KIT Documentation (PR #701)

Sie erhalten nicht oft eine E-Mail von ***@***.******@***.***>. Erfahren Sie, warum dies wichtig ist<https://aka.ms/LearnAboutSenderIdentification>

To be part of the release - Please raise an issue in the sig-release repository (template: KIT Release Checks)
https://github.com/eclipse-tractusx/sig-release/issues/new/choose

-
Reply to this email directly, view it on GitHub<https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/701#issuecomment-1956630407>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7V34IODTH5FIFWJQDBRI7LYUXXUBAVCNFSM6AAAAABDRY3F6CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNJWGYZTANBQG4>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

SebastianBezold commented on docs-kits/kits/Data Governance Kit/page_adoption-view.md:
```diff
@@ -0,0 +1,90 @@
+This work is licensed under the [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/legalcode).
+
+* SPDX-License-Identifier: CC-BY-4.0
+* SPDX-FileCopyrightText: 2021,2022, 2023, 2024 T-Systems International GmbH
```
Are these dates correct? created in 2021?

dg-tsy replied:
Corrected.

SebastianBezold commented on docs-kits/kits/Data Governance Kit/resources/dgk-api-docs.yaml:
Copyright header missing

dg-tsy replied:
This is a YAML file and cannot have any comments or anything else other than data.
I have also renamed this file to openAPI-yaml

SebastianBezold reviewed (commented):
Hi @dg-tsy,

I think the copyright headers, that you added as comment to the `.md` files are unnecessary. We do have a Notice section for that. It does not hurt, but I think it is likely, that the Notice section and copyright header might not be updated consistently

dg-tsy commented on docs-kits/kits/Data Governance Kit/page_changelog.md:
```diff
@@ -0,0 +1,45 @@
+
+### Removed
+
+<sub><sup>(C) 2021,2023 Contributors to the Eclipse Foundation. SPDX-License-Identifier: CC-BY-4.0</sup></sub>
```
Corrected.

dg-tsy commented:
@maximilianong Here is the reply to your feedback comments:
* sidebar navigation for Data Governance Kit added.
* I could not find "Certificate Management Kit" anywhere. Could you please guide me to its correct path?
* Changelog is now updated.
* I thought that "Oparation View" was mandatory. When it's not mandatory, I can remove it in next iteration. 
  There will be no reference implementation.
* If I rename the development view to architecture view, then would I have an architecture view in addition to development view? If not, then where would the API specifications come? I propose to either keep it as it is or add two separate views architecture and development view. Please suggest.

I am still struggling about how to make "kits/Data Governance Kit/resources/openAPI.yaml" as a Tractus-X SwaggerHub API documentation. Any help in this regard is much appreciated.

maximilianong commented:
Please fix the requested changes by 04.03.2024 (EOB) so that we can perform the final checks and prepare the release communication.

@dg-tsy we sent out the following explanation - please make sure you are on the Eclipse mailing list:

For APIs with reference implementation: 
 
Please use (or adapt existing) automated workflows to publish on SwaggerHub. Maintain your openAPI.yaml and the publishing workflow in the repository of the reference implementation. Remove everything related to that from the website repo and only link to SwaggerHub in your KIT from now on.
We utilize a user provided by the Eclipse Foundation to publish on SwaggerHub.
Credentials for this user can be used through the eclipse-tractusx GitHub org secrets and can be utilized in automated workflows. 
Pay special attention to the "swaggerHub" CLI commands.
The location where the openAPI file is maintained may vary. The BPDM example showcases how the application is initiated to download and publish the current file from that location.


      SWAGGERHUB_API_KEY: ${{ secrets.SWAGGERHUB_API_KEY }}
      SWAGGERHUB_USER: ${{ secrets.SWAGGERHUB_USER }}
 
 
Examples:
BPDM Certificate Management: [https://github.com/eclipse-tractusx/bpdm-certificate-management/blob/main/.github/workflows/publish-swagger-hub.yaml]

maximilianong commented:
Operation view:  not mandatory if you do not have something to operate :D please remove it with this PR.

Regarding your suggestion for development view - its ok to go with your approach.

danielmiehle commented:
@dg-tsy can you also fix your current linting issues, please?

![image](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/assets/29955508/cfe7cbe7-17eb-4341-ad41-389916fa4306)

dg-tsy commented:
@maximilianong Following tasks completed:
- Operations View removed, also removed its references from /sidebarsDocsKits.js
- Linting issues resolved
- Added .mdx files for API details, hence changes suggested in [this comment](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/701#issuecomment-1968565806) should not be required.

Please let me know if anything else is required. Looks like all changes done from my side.
